sympy/matrices/tests/test_matrices.py

```diff
@@ -1634,6 +1634,9 @@ def test_diagonalization():
     assert m.is_symmetric()
     assert m.is_diagonalizable()
 
+    M = Matrix([])
+    assert M.diagonal() == Matrix(1, 0, [])
```

Comment:
This test belongs with the other tests for diagonal rather than in the tests for diagonalizable.